integration-tests/src/scenario_utils.rs

```diff
@@ -0,0 +1,236 @@
+        .get_transaction_count(sender.address(), None)
+        .await
+        .expect("cannot get transaction_count");
+
+    for wallet in wallets[0..NUM_TXS].iter() {
+        let tx = TransactionRequest::new()
+            .to(wallet.address())
+            .nonce(nonce)
+            .value(WEI_IN_ETHER * 2u8) // send 2 ETH
+            .from(sender.address());
+        pending_txs.push(
+            wallets[0]
+                .send_transaction(tx, None)
+                .await
+                .expect("cannot send tx"),
+        );
+        nonce = nonce.checked_add(U256::one()).unwrap();
+    }
+
+    sleep(Duration::from_secs(5));
+    block_num = 0;
+    for tx in pending_txs.into_iter() {
+        let receipt = tx.await.expect("cannot confirm tx").unwrap();
+        if block_num == 0 {
+            block_num = receipt.block_number.unwrap().as_u64();
+        } else if block_num != receipt.block_number.unwrap().as_u64() {
+            panic!("The txs are not in a block")
+        }
+    }
+    block_num
+}
+
+/// TransferContext
+pub struct TransferContext {
+    /// Index of `from` wallet
+    pub from_i: usize,
+    /// Index of `to` wallet
+    pub to_i: usize,
+    /// Amount to transfer
+    pub amount: U256,
+    /// Expected result
+    pub expected_result: bool,
+}
+
+impl TransferContext {
+    /// init an instance
+    pub fn new(from_i: usize, to_i: usize, amount: U256, expected_result: bool) -> Self {
+        Self {
+            from_i,
+            to_i,
+            amount,
+            expected_result,
+        }
+    }
+}
+
+/// Build and return erc20 tranfer transaction
+/// TODO(dongchangYoo) Currently this function does not support that an account sends tx twice in a
+/// function.
+pub async fn exchange_token(
+    prov: &Provider<Http>,
+    contract_address: Address,
+    contract_abi: &abi::Contract,
+    wallets: &[WalletType],
+    exchange_map: Vec<TransferContext>,
+    tx_type: usize,
+) -> u64 {
+    let contract = Contract::new(contract_address, contract_abi.clone(), prov.clone());
+
+    let mut pending_txs = Vec::new();
+    for trans_ctx in exchange_map.iter() {
+        let call = contract
+            .method::<_, bool>(
+                "transfer",
+                (wallets[trans_ctx.to_i].address(), trans_ctx.amount),
+            )
+            .expect("cannot construct ERC20 transfer call");
+
+        // Set gas to avoid `eth_estimateGas` call
+        let call = call.gas(100_000);
+
+        // Convert tx to legacy tx if tx_type equals to 0
+        // TODO(dongchangYoo) use constant for tx_type after implementing various tx types
+        let tx = match tx_type {
+            0 => {
+                let call = call.legacy();
+                call.tx
+            }
+            _ => call.tx,
+        };
+
+        pending_txs.push(
+            wallets[trans_ctx.from_i]
+                .send_transaction(tx, None)
+                .await
+                .expect("cannot send token"),
```

Comment:
```suggestion
                .expect("cannot transfer token"),
```

Reply:
Above, you used "cannot send tx" for the error message. Please unify the error message.